doc/sphinx/docs/python/tutorials/using_algorithm.rst

```diff
@@ -0,0 +1,62 @@
+.. py_tutorial_using_algorithm
+
+Use of the class :class:`~pygmo.core.algorithm`
+===============================================
+
+The :class:`~pygmo.core.algorithm` class represents a generic optimization
+algorithm. The user codes the details of such an algorithm in a separate class (the
+user-defined algorithm, or UDA) which is then passed to :class:`~pygmo.core.algorithm`
+that provides a single unified interface.
+
+.. note::  The User Defined Algorithms (UDAs) are optimization algorithms (coded by the user) that can
+           be used to build a pygmo object of type :class:`~pygmo.core.algorithm`
+
+Some UDAs (optimization algorithms) are already provided with pygmo and we refer to them as pygmo UDAs.
```

Comment:
The reason is 1) I like it more, 2) it allows to be able to import PyGMO and pygmo as separate projects (this is handy for me to check quickly on the algo perf and for the users to have both installed at the same time)